packages/core/test-helpers/src/websocket.ts

```diff
@@ -45,6 +45,10 @@ export const mockWebSocketFlow = async (
     let currentExchange = 0
     server.on('connection', async (connection) => {
       connection.on('message', (received) => {
+        if (!flow[currentExchange]) {
+          console.warn(`Received value is not in flow. Received: ${received}`)
+          return
+        }
```

Comment:
Is this to resolve the issues we're seeing with other websocket tests? Do you know the underlying cause?

Reply:
This was one of the issues. Not handling this will throw errors and some tests, not necessarily this one, would fail. This can happen because the way the websocket integration test framework is set up is very time sensitive so without having this check 2 out of 10 tests might fail.